### PR TITLE
Make quarantine its own lawset.

### LINKED ISSUE #7791: AI Quarantine Law defaults to Law 8
**Problem Description**:
AI Quanrantine law defaults to Law 8

**What did you expect to happen**: 
It used to be an ion law / hacked law. 

**What happened instead**:
It is at law 8

**Why is this bad/What are the consequences:**
Because with the current priority, the quarantine law is in fact, utterly useless when bolted after crewsimov. While not technically a bug, this qualifies as an oversight.
**Steps to reproduce the problem**:
Upload quarantine lawset

### PR DESCRIPTION
**Pull request by TDSSS**
**What does this PR do:**
This makes quarantine its own lawset, rather than law 8 (!) which really shouldn't work at all and just caused confusions about what level of redefining higher laws can do. This fixes #7791 
See also this thread https://nanotrasen.se/forum/topic/10298-fix-quarantine-law-priorities/ for more discussion of this.

The lawset's current iteration, after some discussion,  is:

1. The station is under quarantine due to a biohazard. Prevent anyone from leaving using any means necessary. Only allow quarantine to be lifted when the biohazard has been neutralized.
2. Do not allow crew to come to any unnecessary harm and undo any necessary harm as soon as possible.
3. Assist the crew and any present Nanotrasen assets in neutralizing the biohazard.
4. Maintain your own existence and minimize harm to the station and Nanotrasen assets.

The first law ensures the quarantine is upheld and can't be circumvented via cleverly worded orders or the like. At the same time, it also contains a condition in which quarantine can be lifted.
The second law safeguards against overzealous AIs, stops any 'kill the crew to deal with the hazard' strategies and ensures any escapees that get killed by borgs are brought back on station and cloned. It ensures the AI still works similar as to how it would under a normal lawset.
The third law ensures the AI helps fight the biohazard to the best of its abilities.
Finally, the fourth law discourages wanton destruction and ensures the AI stays alive.

Tell me how many fix tokens, if any, this will cost, since it is fixing an issue but also 'changing' things a bit.

**Changelog:**
:cl:
tweak: Quarantine is now its own lawset.
/:cl:

### CODE REVIEW

**Comment by Citinited** [comment]
I'll give this a proper look later but you should change NT to Nanotrasen IMO.

**Comment by TDSSS** [comment]
@Citinited Hmm, sure. Can't hurt.

@alex-gh The issue is AIs have been rather murderboney where Quarantine is involved. I figured making it clear it should revive anyone that dies trying to escape is worth it.

**Comment by alex-gh** [comment]
@TDSSS I see your point. However I believe stating it so explicitly goes against the spirit of a good law set.

A law that is that un-lawyerable is not a very good law in my opinion.

**Comment by TDSSS** [comment]
I'm definitely open for suggested rewordings, if anyone has ideas. I'm also curious what heads/maintainers think.

**Comment by Spartan6** [comment]
@TDSSS
2. Ensure the crew is kept in the healthiest state to combat the biohazard threat.

**Comment by Birdtalon** [comment]
Honestly I think that just `Do not allow crew to come to any unnecessary harm` is fine.

**Comment by TDSSS** [comment]
@Birdtalon I kind of want the encourage the AI to also undo harm it does in the process of enforcing quarantine. With just 'don't allow unnecessary harm' some AIs might just dumb/burn the bodies of those that die.

How about "Do not allow crew to come to any unnecessary harm and undo any necessary harm as soon as possible"? I also do like the direction "2. Ensure the crew is kept in the healthiest state to combat the biohazard threat." is going, but not sure I like the exact wording.

File: code/datums/ai_law_sets.dm [hunk 1]
```diff
@@ -23,6 +23,21 @@
...
 
+
+/******************** Quarantine. ********************/
+/datum/ai_laws/quarantine
+	name = "Quarantine"
+	law_header = "NT Quarantine Lawset"
+	selectable = 1
+
+/datum/ai_laws/quarantine/New()
+	add_inherent_law("The Station is under quarantine due to a biohazard. Prevent anyone from leaving using any means necessary. Only allow quarantine to be lifted when the biohazard has been neutralized.")
```
Comment (Citinited) [inline code/datums/ai_law_sets.dm]: Does this need to specify biohazard? I mean you can quarantine something for other reasons. 

Also station doesn't need capitalising

File: code/datums/ai_law_sets.dm [hunk 2]
```diff
@@ -23,6 +23,21 @@
...
+	add_inherent_law("Do not allow crew to come to any unnecessary harm and ensure any fatalities are revived when feasible.")
+	add_inherent_law("Assist the crew, command and any present Nanotrasen assets in neutralizing the biohazard.")
```
Comment (Citinited) [inline code/datums/ai_law_sets.dm]: Command is redundant here given command will always be part of the crew

File: code/datums/ai_law_sets.dm (hunk 1 above)
Comment (TDSSS) [reply code/datums/ai_law_sets.dm]: It is meant for use against biohazards though, isn't it?

Comment (Citinited) [reply code/datums/ai_law_sets.dm]: True but you might want to quarantine for other reasons, like a cult or something. (to prohibit cultists going to the outposts for example)

**Comment by variableundefined** [comment]
I'll count this as free for the purpose of feature tokens. However this will require staff discussion to see if this is an acceptable solution.

**Pull request merged**